api/tests/src/api/applications/test_application_routes.py

```diff
@@ -3287,29 +3287,202 @@ def test_application_form_inclusion_update_application_form_not_found(
+        select(ApplicationForm).where(
+            ApplicationForm.application_id == application_id,
+            ApplicationForm.competition_form_id == competition_form.competition_form_id
+        )
+    ).scalar_one_or_none()
+
+    assert application_form is not None
+    assert application_form.application_response["opportunity_number_field"] == "TEST-OPP-123"
+    assert application_form.application_response["opportunity_title_field"] == "Test Opportunity Title"
+
+
+def test_application_form_update_with_pre_population(
+    client, enable_factory_create, db_session, user, user_auth_token
+):
+    """Test that updating an application form triggers pre-population of new fields"""
+    # Create application
+    opportunity = OpportunityFactory.create(
+        opportunity_number="UPDATE-OPP-456",
+        opportunity_title="Updated Opportunity Title"
+    )
+    
+    application = ApplicationFactory.create()
+    application.competition.opportunity = opportunity
+
+    # Create a form with pre-population rules
+    form_rule_schema = {
+        "opportunity_number_field": {
+            "gg_pre_population": {"rule": "opportunity_number"}
+        },
+        "user_input_field": {
+            # No rule - user can modify this
+        }
+    }
+    
+    form = FormFactory.create(
+        form_json_schema=SIMPLE_JSON_SCHEMA,
+        form_rule_schema=form_rule_schema
+    )
+
+    competition_form = CompetitionFormFactory.create(
+        competition=application.competition,
+        form=form,
+    )
+
+    # Associate user with application
+    ApplicationUserFactory.create(user=user, application=application)
+
+    application_id = str(application.application_id)
+    form_id = str(competition_form.form_id)
+    
+    # Update with user data - should trigger pre-population
+    request_data = {
+        "application_response": {
+            "user_input_field": "User provided data"
+        }
+    }
+
+    response = client.put(
+        f"/alpha/applications/{application_id}/forms/{form_id}",
+        json=request_data,
+        headers={"X-SGG-Token": user_auth_token},
+    )
+
+    assert response.status_code == 200
+    assert response.json["message"] == "Success"
+
+    # Verify application form was updated with both user data and pre-populated values
+    application_form = db_session.execute(
+        select(ApplicationForm).where(
+            ApplicationForm.application_id == application.application_id,
+            ApplicationForm.competition_form_id == competition_form.competition_form_id,
+        )
+    ).scalar_one_or_none()
+
+    assert application_form is not None
+    # Should have user data
+    assert application_form.application_response["user_input_field"] == "User provided data"
+    # Should also have pre-populated data
+    assert application_form.application_response["opportunity_number_field"] == "UPDATE-OPP-456"
+
+
+def test_application_form_update_preserves_user_changes_with_pre_population(
```

Comment:
I think I'd reword this test name to:
```suggestion
def test_application_form_update_overwrites_user_changes_with_pre_population(
```